File: Java/Longest Nice Subarray.java
public class Solution {

    /**
     * Time:   O(n)
     * Memory: O(1)
     */
    public int longestNiceSubarray(int[] nums) {
        int longest = 0, used_bits = 0, j = 0;

        for (int i = 0; i < nums.length; ++i) {
            while ((used_bits & nums[i]) != 0)
                used_bits ^= nums[j++];

            used_bits |= nums[i];
            longest = Math.max(longest, i - j + 1);
        }

        return longest;
    }
}

File: Python/Longest Nice Subarray.py
from typing import List


class Solution:
    """
    Time:   O(n)
    Memory: O(1)
    """

    def longestNiceSubarray(self, nums: List[int]) -> int:
        longest = used_bits = j = 0

        for i, num in enumerate(nums):
            while used_bits & num != 0:
                used_bits ^= nums[j]
                j += 1

            used_bits |= num
            longest = max(longest, i - j + 1)

        return longest
